### ISSUE TITLE
Missing required argument #1 when update

### ISSUE THREAD

**Issue by ghaiklor** [npm/npm#11085]
Stack trace from console:

```shell
ghaiklor@ghaiklor-MacBook-Pro ~/L/Projects> sudo npm -g update
npm ERR! Darwin 15.2.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "-g" "update"
npm ERR! node v5.4.0
npm ERR! npm  v3.5.3
npm ERR! code EMISSINGARG

npm ERR! typeerror Error: Missing required argument #1
npm ERR! typeerror     at module.exports (/usr/local/lib/node_modules/npm/lib/install/mutate-into-logical-tree.js:12:3)
npm ERR! typeerror     at /usr/local/lib/node_modules/npm/lib/outdated.js:78:5
npm ERR! typeerror     at /usr/local/lib/node_modules/npm/lib/outdated.js:62:20
npm ERR! typeerror     at /usr/local/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:186:9
npm ERR! typeerror     at f (/usr/local/lib/node_modules/npm/node_modules/once/once.js:17:25)
npm ERR! typeerror     at /usr/local/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:152:20
npm ERR! typeerror     at f (/usr/local/lib/node_modules/npm/node_modules/once/once.js:17:25)
npm ERR! typeerror     at then (/usr/local/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:121:22)
npm ERR! typeerror     at /usr/local/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:81:20
npm ERR! typeerror     at fs.js:1657:23
npm ERR! typeerror This is an error with npm itself. Please report this error at:
npm ERR! typeerror     <http://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/ghaiklor/Library/Projects/npm-debug.log
ghaiklor@ghaiklor-MacBook-Pro ~/L/Projects>
```

npm-debug.log

```
ghaiklor@ghaiklor-MacBook-Pro ~/L/Projects> cat npm-debug.log
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', '-g', 'update' ]
2 info using npm@3.5.3
3 info using node@v5.4.0
4 verbose stack Error: Missing required argument #1
4 verbose stack     at module.exports (/usr/local/lib/node_modules/npm/lib/install/mutate-into-logical-tree.js:12:3)
4 verbose stack     at /usr/local/lib/node_modules/npm/lib/outdated.js:78:5
4 verbose stack     at /usr/local/lib/node_modules/npm/lib/outdated.js:62:20
4 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:186:9
4 verbose stack     at f (/usr/local/lib/node_modules/npm/node_modules/once/once.js:17:25)
4 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:152:20
4 verbose stack     at f (/usr/local/lib/node_modules/npm/node_modules/once/once.js:17:25)
4 verbose stack     at then (/usr/local/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:121:22)
4 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:81:20
4 verbose stack     at fs.js:1657:23
5 verbose cwd /Users/ghaiklor/Library/Projects
6 error Darwin 15.2.0
7 error argv "/usr/local/bin/node" "/usr/local/bin/npm" "-g" "update"
8 error node v5.4.0
9 error npm  v3.5.3
10 error code EMISSINGARG
11 error typeerror Error: Missing required argument #1
11 error typeerror     at module.exports (/usr/local/lib/node_modules/npm/lib/install/mutate-into-logical-tree.js:12:3)
11 error typeerror     at /usr/local/lib/node_modules/npm/lib/outdated.js:78:5
11 error typeerror     at /usr/local/lib/node_modules/npm/lib/outdated.js:62:20
11 error typeerror     at /usr/local/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:186:9
11 error typeerror     at f (/usr/local/lib/node_modules/npm/node_modules/once/once.js:17:25)
11 error typeerror     at /usr/local/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:152:20
11 error typeerror     at f (/usr/local/lib/node_modules/npm/node_modules/once/once.js:17:25)
11 error typeerror     at then (/usr/local/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:121:22)
11 error typeerror     at /usr/local/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:81:20
11 error typeerror     at fs.js:1657:23
12 error typeerror This is an error with npm itself. Please report this error at:
12 error typeerror     <http://github.com/npm/npm/issues>
13 verbose exit [ 1, true ]
ghaiklor@ghaiklor-MacBook-Pro ~/L/Projects>
```

**Comment by YannBertrand** [comment]
Same issue here using npm@3.5.3, node@4.2.3 (nvm@0.29.0)

**Comment by halhenke** [comment]
same when doing `npm -g outdated`:
node@5.4.1
npm@3.5.3

**Comment by halhenke** [comment]
In my case it was due to having a global package symlink whose target had been deleted.

**Comment by pdelorme** [comment]
same issue with ubuntu 14.04 
```
➜  ~  sudo npm update -g 
npm ERR! Linux 3.13.0-76-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "update" "-g"
npm ERR! node v4.2.6
npm ERR! npm  v3.5.3
npm ERR! code EMISSINGARG

npm ERR! typeerror Error: Missing required argument #1
npm ERR! typeerror     at module.exports (/usr/lib/node_modules/npm/lib/install/mutate-into-logical-tree.js:12:3)
```

**Comment by ekumachidi** [comment]
npm WARN deprecated lodash-node@2.4.1: This package is no longer maintained. See its readme for upgrade details.
npm ERR! Linux 3.13.0-76-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "install" "-g" "ember-cli"
npm ERR! node v5.5.0
npm ERR! npm  v3.3.12

npm ERR! Maximum call stack size exceeded
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

**Comment by bdukes** [comment]
I've just hit this error on npm 3.7.1, running `npm outdated`.  http://pastebin.com/dFN93EF4

**Comment by alice-em** [comment]
`0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/bin/npm', 'update', '-g' ]
2 info using npm@3.7.5
3 info using node@v5.7.1
4 verbose stack Error: Missing required argument #1
4 verbose stack     at module.exports (/usr/lib/node_modules/npm/lib/install/mutate-into-logical-tree.js:12:3)
4 verbose stack     at /usr/lib/node_modules/npm/lib/outdated.js:78:5
4 verbose stack     at /usr/lib/node_modules/npm/lib/outdated.js:62:20
4 verbose stack     at /usr/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:186:9
4 verbose stack     at f (/usr/lib/node_modules/npm/node_modules/once/once.js:17:25)
4 verbose stack     at then (/usr/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:166:22)
4 verbose stack     at f (/usr/lib/node_modules/npm/node_modules/once/once.js:17:25)
4 verbose stack     at /usr/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:152:20
4 verbose stack     at f (/usr/lib/node_modules/npm/node_modules/once/once.js:17:25)
4 verbose stack     at then (/usr/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:121:22)
5 verbose cwd /home/alicemod
6 error Linux 4.4.3-300.fc23.x86_64
7 error argv "/usr/bin/node" "/bin/npm" "update" "-g"
8 error node v5.7.1
9 error npm  v3.7.5
10 error code EMISSINGARG
11 error typeerror Error: Missing required argument #1
11 error typeerror     at module.exports (/usr/lib/node_modules/npm/lib/install/mutate-into-logical-tree.js:12:3)
11 error typeerror     at /usr/lib/node_modules/npm/lib/outdated.js:78:5
11 error typeerror     at /usr/lib/node_modules/npm/lib/outdated.js:62:20
11 error typeerror     at /usr/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:186:9
11 error typeerror     at f (/usr/lib/node_modules/npm/node_modules/once/once.js:17:25)
11 error typeerror     at then (/usr/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:166:22)
11 error typeerror     at f (/usr/lib/node_modules/npm/node_modules/once/once.js:17:25)
11 error typeerror     at /usr/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:152:20
11 error typeerror     at f (/usr/lib/node_modules/npm/node_modules/once/once.js:17:25)
11 error typeerror     at then (/usr/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:121:22)
12 error typeerror This is an error with npm itself. Please report this error at:
12 error typeerror     <http://github.com/npm/npm/issues>
13 verbose exit [ 1, true ]`

I've been getting the same error since I updated node using the nodesource repo on Fedora.  `npm -v` returns v3.6.0, even if it says 3.7.5 above.

**Comment by ghaiklor** [comment]
@alice-em wrap your code in ` \`\`\` `, please.

**Comment by alice-em** [comment]
D'oh, I didn't realize I had to do it line by line.  I fail at my markdown-kata.  *the more you know

I did a complete reinstallation of node and npm on my linux machine, and that seems to have solved the issue, for the time being.

**Comment by ghaiklor** [comment]
@alice-em the same. it fixes after cleaning symlinks.

**Comment by iarna** [comment]
Good news! This was fixed in [v3.10.2](https://github.com/npm/npm/releases/tag/v3.10.2) with https://github.com/npm/npm/pull/13121.
